api/v1alpha1/clusterexternaldnsconfiguration_types.go

```diff
@@ -0,0 +1,94 @@
+package v1alpha1
+
+import (
+	"github.com/Azure/aks-app-routing-operator/api"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+//+kubebuilder:resource:scope=Cluster,shortName=cedc
+
+// ClusterExternalDNSConfiguration allows users to specify desired the state of a cluster-scoped ExternalDNS configuration and includes information about the state of their resources in the form of Kubernetes events.
+type ClusterExternalDNSConfiguration struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ClusterExternalDNSConfigurationSpec   `json:"spec,omitempty"`
+	Status ClusterExternalDNSConfigurationStatus `json:"status,omitempty"`
+}
+
+func (c *ClusterExternalDNSConfiguration) GetCondition(conditionType string) *metav1.Condition {
+	return meta.FindStatusCondition(c.Status.Conditions, conditionType)
+}
+
+func (c *ClusterExternalDNSConfiguration) GetConditions() *[]metav1.Condition {
+	return &c.Status.Conditions
+}
+
+func (c *ClusterExternalDNSConfiguration) GetGeneration() int64 {
+	return c.Generation
+}
+
+func (c *ClusterExternalDNSConfiguration) SetCondition(condition metav1.Condition) {
+	api.VerifyAndSetCondition(c, condition)
+}
+
+// ClusterExternalDNSConfigurationSpec allows users to specify desired the state of a cluster-scoped ExternalDNS configuration.
+type ClusterExternalDNSConfigurationSpec struct {
+	// TenantID is the ID of the Azure tenant where the DNS zones are located.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Format:=uuid
+	// +kubebuilder:validation:Pattern=`[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}`
+	TenantID string `json:"tenantID"`
+
+	// DNSZoneResourceIDs is a list of Azure Resource IDs of the DNS zones that the ExternalDNS controller should manage. These should be in the same resource group and be of the same type (public or private).
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems:=1
+	// +kubebuilder:validation:MaxItems:=20
```

Comment:
where does this come from?

Reply:
why 20?

Reply:
it's arbitrary at the moment. @sabbour do we have a better figure for what our users want?

Post unblocking of large testing framework I can also look to see how large we can make one deployment - externaldns doesn't document any limitations

Reply:
externaldns scales on records not domains. 20 seems fine for now but would probably be good to document this in comments for now